Backend prediction functions

furkanaydar opened the pull request:
### Description
What does this implement/fix? Explain your changes.

### Related Issues
...

### Other Comments
...

### Checklist
- [ ] Code runs
- [ ] Created tests (if applicable)
- [ ] All tests passing
- [ ] Extended the README and the documentation (if applicable)

alpercakan commented:
I would suggest using a single binary value to represent the prediction to eliminate redundancy, but it is not crucial nor urgent. So, ✅.

furkanaydar commented:
@alpercakan I realized that I was just creating a new prediction with same key value, changed it with an update. Also changed vote field to binary in request. Thanks for advice.
